src/run_program.rs

```diff
@@ -41,8 +47,9 @@ pub struct RunProgramContext<'a, T: Allocator> {
     apply_kw: u8,
     operator_lookup: Box<dyn OperatorHandler<T>>,
     pre_eval: Option<PreEval<T>>,
+    posteval_stack: Vec<Box<PostEval<T>>>,
     val_stack: Vec<T::Ptr>,
-    op_stack: Vec<Box<RpcOperator<T>>>,
+    op_stack: Vec<Operation>,
```

Comment:
You might get the same benefit (and reduce complexity) if you just unbox the functions. I didn't think about moving `PostEval` to its own stack, and it requires boxing, but the others all seem static.